<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>445</width>
    <height>467</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>RaceExample</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0" colspan="2">
     <widget class="QTextEdit" name="te_debug">
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="4" column="0" colspan="2">
     <widget class="QPushButton" name="pb_clear">
      <property name="text">
       <string>Очистить</string>
      </property>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="QLabel" name="lb_initNum">
      <property name="text">
       <string>Удваиваемое число:</string>
      </property>
     </widget>
    </item>
    <item row="5" column="0" colspan="2">
     <widget class="QPushButton" name="pb_start">
      <property name="text">
       <string>Старт</string>
      </property>
     </widget>
    </item>
    <item row="3" column="1">
     <widget class="QSpinBox" name="sb_initNum">
      <property name="maximum">
       <number>1000000000</number>
      </property>
     </widget>
    </item>
    <item row="2" column="0" colspan="2">
     <widget class="QGroupBox" name="gb_synch">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="title">
       <string>Синхронизация потоков</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QRadioButton" name="rb_synchOff">
         <property name="text">
          <string>Без синхронизации</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="rb_mutexOn">
         <property name="text">
          <string>Включить мьютексы</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="rb_qtConcur">
         <property name="text">
          <string>Qt Concurrent</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>445</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <tabstops>
  <tabstop>te_debug</tabstop>
  <tabstop>sb_initNum</tabstop>
  <tabstop>pb_clear</tabstop>
  <tabstop>pb_start</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>pb_clear</sender>
   <signal>clicked()</signal>
   <receiver>te_debug</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>198</x>
     <y>262</y>
    </hint>
    <hint type="destinationlabel">
     <x>306</x>
     <y>117</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
